### PR TITLE
Additional sniffs and tests

### PR DESCRIPTION
**Pull request by webimpress**
@xtreamwayz this is the second part I was talking about

We need to add some test for sniffs - I marked them with `@todo` tag.

Again, documentation is not updated, I can do it later or feel free to do it :)

### CODE REVIEW

File: src/ZendCodingStandard/ruleset.xml [hunk 1]
```diff
@@ -247,17 +292,13 @@
...
-    </rule>
+    <rule ref="WebimpressCodingStandard.Files.DeclareStrictTypes"/>
+    <!-- Format of strict type declaration -->
+    <rule ref="WebimpressCodingStandard.PHP.DeclareStrictTypes"/>
     <!-- Forbid useless parentheses -->
     <rule ref="WebimpressCodingStandard.Formatting.UnnecessaryParentheses"/>
-    <!-- Forbid useless semicolon `;` -->
-    <rule ref="SlevomatCodingStandard.PHP.UselessSemicolon"/>
+    <!-- Forbid redundant semicolons -->
+    <rule ref="WebimpressCodingStandard.PHP.RedundantSemicolon"/>
```
Comment (Ocramius) [inline src/ZendCodingStandard/ruleset.xml]: Any reason for replacing these Slevomat tools with Webimpress ones, besides ownership?

File: test/fixed/concatenation-spacing.php [hunk 2]
```diff
@@ -20,6 +20,6 @@
           . '2'
       . '3';
 
-$string = '1' .
-          '2' .
-          '3';
+$string = '1'
+          . '2'
+          . '3';
```
Comment (xtreamwayz) [inline test/fixed/concatenation-spacing.php]: Shouldn't these two be aligned with the equal sign?

```
$string = '1'
        . '2'
        . '3';
```

Comment (webimpress) [reply test/fixed/concatenation-spacing.php]: Maybe it should but we don't have any sniff like that... And what if concatenation is in let say `sprintf`:

```php
throw new \Exception(sprintf(
    'This is very very very long error message'
    . ' with some exception: %s',
    $e->getMessage()
));
```

There is so many cases. ScopeIndent from PHP_CodeSniffer doesn't include so many cases and my is super strict, and controversial in some places ;-) We can try, but ... hah

**Comment by xtreamwayz** [comment]
Thanx @webimpress

**Pull request merged**